tools/graphiteindex/src/main/java/com/criteo/biggraphite/graphiteindex/LuceneIndexSearcher.java

```diff
@@ -138,14 +142,15 @@ public UnfilteredRowIterator next() {
         }
 
         /**
-         * TODO(p.boddu): Lot of heavy lifting here for each query. Optimize!
+         * SearcherManager facilitates safely sharing IndexSearcher instances across multiple thread.
+         * TODO(p.boddu): Confirm that calling FSDirectory.open() on every search call doesn't hog system resources.
```

Comment:
As far as I can remember, I believe we should have the search manager etc. in the class and it will periodically reopen things (possibly with some task). I did not do it during the initial phase but we should work on that next :-)